internal/k8s/appprotectdos/app_protect_dos_configuration.go

```diff
@@ -0,0 +1,271 @@
+	resNsName := GetNsName(policyObj)
+	policy, err := createAppProtectDosPolicyEx(policyObj)
+	if err != nil {
+		ci.DosPolicies[resNsName] = policy
+		return append(changes, Change{Op: Delete, Resource: policy}),
+			append(problems, Problem{Object: policyObj, Reason: "Rejected", Message: err.Error()})
+	}
+	ci.DosPolicies[resNsName] = policy
+	return append(changes, Change{Op: AddOrUpdate, Resource: policy}), problems
+}
+
+// AddOrUpdateDosLogConf adds or updates App Protect Dos Log Configuration to App Protect Dos Configuration
+func (ci *ConfigurationImpl) AddOrUpdateDosLogConf(logconfObj *unstructured.Unstructured) (changes []Change, problems []Problem) {
+	resNsName := GetNsName(logconfObj)
+	logConf, err := createAppProtectDosLogConfEx(logconfObj)
+	ci.DosLogConfs[resNsName] = logConf
+	if err != nil {
+		return append(changes, Change{Op: Delete, Resource: logConf}),
+			append(problems, Problem{Object: logconfObj, Reason: "Rejected", Message: err.Error()})
+	}
+	return append(changes, Change{Op: AddOrUpdate, Resource: logConf}), problems
+}
+
+// GetAppDosResource returns a pointer to an App Protect Dos resource
+func (ci *ConfigurationImpl) GetAppDosResource(kind, key string) (*unstructured.Unstructured, error) {
+	switch kind {
+	case DosPolicyGVK.Kind:
+		if obj, ok := ci.DosPolicies[key]; ok {
+			if obj.IsValid {
+				return obj.Obj, nil
+			}
+			return nil, fmt.Errorf(obj.ErrorMsg)
+		}
+		return nil, fmt.Errorf("App Protect Dos Policy %s not found", key)
+	case DosLogConfGVK.Kind:
+		if obj, ok := ci.DosLogConfs[key]; ok {
+			if obj.IsValid {
+				return obj.Obj, nil
+			}
+			return nil, fmt.Errorf(obj.ErrorMsg)
+		}
+		return nil, fmt.Errorf("App Protect DosLogConf %s not found", key)
+	}
+	return nil, fmt.Errorf("Unknown App Protect Dos resource kind %s", kind)
+}
+
+// DeleteDosPolicy deletes an App Protect Policy from App Protect Dos Configuration
+func (ci *ConfigurationImpl) DeleteDosPolicy(key string) (changes []Change, problems []Problem) {
+	if _, has := ci.DosPolicies[key]; has {
+		change := Change{Op: Delete, Resource: ci.DosPolicies[key]}
+		delete(ci.DosPolicies, key)
+		return append(changes, change), problems
+	}
+	return changes, problems
+}
+
+// DeleteDosLogConf deletes an App Protect Dos Log Configuration from App Protect Dos Configuration
+func (ci *ConfigurationImpl) DeleteDosLogConf(key string) (changes []Change, problems []Problem) {
+	if _, has := ci.DosLogConfs[key]; has {
+		change := Change{Op: Delete, Resource: ci.DosLogConfs[key]}
+		delete(ci.DosLogConfs, key)
+		return append(changes, change), problems
+	}
+	return changes, problems
+}
+
+// FakeConfiguration holds representations of fake App Protect Dos cluster resources
+type FakeConfiguration struct {
+	DosPolicies map[string]*DosPolicyEx
+	DosLogConfs map[string]*DosLogConfEx
+}
+
+// NewFakeConfiguration creates a new App Protect Dos Configuration
+func NewFakeConfiguration() Configuration {
```

Comment:
it doesn't look like FakeConfiguration is used anywhere in the code. If it is the case, than better to remove it

Reply:
Used now in the new unit test: TestAddBadosPolicyRefs